src/classes/RD2_OpportunityEvaluationService_TEST.cls

```diff
@@ -238,17 +245,212 @@ private class RD2_OpportunityEvaluationService_TEST {
+    * no installment should be created on inserted of RD and during nightly batch job
+    */
+    @IsTest
+    private static void shouldNotCreateOppOnInsertWhenDiableAllInstallment() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Map<String, Object> rdSettingMap = new Map<String, Object> {
+            'InstallmentOppAutoCreateOption__c' => OPTION_NAME_DISABLE_ALL
+        };
+
+        RD2_Settings_TEST.setUpConfiguration(rdSettingMap);
+
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder(getContact()).build();
+
+        Test.startTest();
+        insert rd;
+        Test.stopTest();
+        
+        List<Opportunity> opps = oppGateway.getRecords(rd);
+
+        System.assertEquals(0, opps.size(),
+            'The installment opportunity should not be created when the Auto Creation Setting is' + OPTION_NAME_DISABLE_ALL);
+    }
+
+    /**
+    * @description When Installment Auto Creation Setting is set to Disable All Installments, 
+    * no installment should be created on inserted of RD and during nightly batch job
+    */
+    @IsTest
+    private static void shouldNotCreateOppDuringBatchRunWhenDiableAllInstallment() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Map<String, Object> rdSettingMap = new Map<String, Object> {
+            'InstallmentOppAutoCreateOption__c' => OPTION_NAME_DISABLE_ALL
+        };
+
+        RD2_Settings_TEST.setUpConfiguration(rdSettingMap);
+
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder(getContact()).build();
+        insert rd;
+        
+        Test.startTest();
+        Database.executeBatch(new RD2_OpportunityEvaluation_BATCH());
+        Test.stopTest();
+        
+        List<Opportunity> opps = oppGateway.getRecords(rd);
+
+        System.assertEquals(0, opps.size(),
+            'The installment opportunity should not be created when the Auto Creation Setting is' + OPTION_NAME_DISABLE_ALL);
+    }
+
+    /**
+    * @description When installment Creation Setting is Set to disable First On Create, a new isntallment opp
+    * should be created when the RD is updated and there is already a Installment Opp existed with CloseDate before today
+    */
+    @IsTest
+    private static void shouldCreateOppOnUpdateOfRDWhenClosedOppExistAndDisableFirstInstallment() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Map<String, Object> rdSettingMap = new Map<String, Object> {
+            'InstallmentOppAutoCreateOption__c' => OPTION_NAME_DISABLE_FIRST
+        };
+
+        RD2_Settings_TEST.setUpConfiguration(rdSettingMap);
+        
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder(getContact()).build();
+        insert rd;
+
+        insert getOpportunityBuilder(rd)
+            .withName()
+            .withCloseDate(Date.Today().addDays(-10))
```

Comment:
Let's match the installment Opp close date with a Next Donation Date that should have occurred in past, ie `START_DATE` or `START_DATE.addMonths(1)` for example.